#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_2d_localizer"

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "", True)
gen.add("corr_xx", double_t, 0, "", 0.0, 0.0, 100.0)
gen.add("corr_xy", double_t, 0, "", 0.0, 0.0, 100.0)
gen.add("corr_xa", double_t, 0, "", 0.0, 0.0, 100.0)
gen.add("corr_yy", double_t, 0, "", 0.0, 0.0, 100.0)
gen.add("corr_ya", double_t, 0, "", 0.0, 0.0, 100.0)
gen.add("corr_aa", double_t, 0, "", 1.0, 0.0, 100.0)
gen.add("corr_magnitude", double_t, 0, "", 0.001, 0.0, 100.0)
gen.add("alpha_filter", double_t, 0, "", 0.75, 0.0, 1.0)
gen.add("max_lin_vel", double_t, 0, "", 0.5, 0.0, 100.0)
gen.add("max_ang_vel", double_t, 0, "", 0.7, 0.0, 100.0)

exit(gen.generate(PACKAGE_NAME, "squirrel_2d_localizer", "TwistCorrection"))